sunpy/coordinates/great_arc.py

```diff
@@ -0,0 +1,121 @@
+#
+# Calculates the co-ordinates along great arcs between two specified points
+# which are assumed to be on disk.
+#
+from __future__ import absolute_import, division, print_function
+
+import numpy as np
+from astropy.coordinates import SkyCoord
+import sunpy.coordinates
+
+
+def great_arc(start_point, end_point, center=None, number_points=100):
+    """
+    Calculate a user-specified number of points on a great arc between a start
+    and end point on a sphere.
+
+    Parameters
+    ----------
+    start_point : `~astropy.coordinates.SkyCoord`
+        Start point.
+
+    end_point : `~astropy.coordinates.SkyCoord`
+        End point.
+
+    center : `~astropy.coordinates.SkyCoord`
+        Center of the sphere.
+
+    number_points : int
+        Number of points along the great arc.
+
+    Returns
+    -------
+    arc : `~astropy.coordinates.SkyCoord`
+        Co-ordinates along the great arc in the co-ordinate frame of the
+        start point.
+
+    Example
+    -------
+    >>> from astropy.coordinates import SkyCoord
+    >>> import astropy.units as u
+    >>> import sunpy.coordinates
+    >>> from sunpy.coordinates import great_arc
+    >>> import sunpy.map
+    >>> from sunpy.data.sample import AIA_171_IMAGE
+    >>> m = sunpy.map.Map(AIA_171_IMAGE)
+    >>> a = SkyCoord(600*u.arcsec, -600*u.arcsec, frame=m.coordinate_frame)
+    >>> b = SkyCoord(-100*u.arcsec, 800*u.arcsec, frame=m.coordinate_frame)
+    >>> v = great_arc(a, b)
+    """
+    start_point_unit = start_point.transform_to('heliocentric').cartesian.xyz.unit
+    if center is None:
+        c = SkyCoord(0*start_point_unit, 0*start_point_unit, 0*start_point_unit, frame='heliocentric')
+    a_cartesian = start_point.transform_to('heliocentric').cartesian.xyz.to(start_point_unit).value
+    b_cartesian = end_point.transform_to('heliocentric').cartesian.xyz.to(start_point_unit).value
+    c_cartesian = c.transform_to('heliocentric').cartesian.xyz.to(start_point_unit).value
+
+    # Calculate the points along the great arc.
+    v_cartesian = calculate_great_arc(a_cartesian, b_cartesian, c_cartesian, number_points)*start_point_unit
+
+    # Transform the great arc back into the input frame.
+    return SkyCoord(v_cartesian[:, 0], v_cartesian[:, 1], v_cartesian[:, 2],
+                    frame='heliocentric', observer=start_point.observer).transform_to(start_point.frame)
+
+
+def calculate_great_arc(a_cartesian, b_cartesian, c_cartesian, number_points):
```

Comment:
Should be private as it does not take units therefore should not be user facing as per SEP.